mlflow/sklearn/utils.py

```diff
@@ -219,18 +239,23 @@ def _get_classifier_metrics(fitted_estimator, fit_args, fit_kwargs):
         classifier_metrics.extend(
             [
                 _SklearnMetric(
-                    name=_METRICS_PREFIX + "log_loss",
+                    name=_TRAINING_PREFIX + "log_loss",
                     function=sklearn.metrics.log_loss,
                     arguments=dict(y_true=y_true, y_pred=y_pred_proba, sample_weight=sample_weight),
                 ),
             ]
         )
 
         if _is_metric_supported("roc_auc_score"):
+            # For binary case, the parameter `y_score` expect scores must be
+            # the scores of the class with the greater label.
+            if y_pred_proba.shape[1] == 2:
```

Comment:
I still don't think we want to assume that `y_pred_proba` has a `shape` attribute. It doesn't necessarily have to be a numpy array or a DataFrame. Could we instead check the length of the first element of `y_pred_proba`? This way, we're only assuming that `y_pred_proba` is an arbitrary 2D array, rather than a numpy array.